cpp/tests/test_odesecir.cpp

```diff
@@ -1177,5 +1178,206 @@ TEST(Secir, read_population_data_one_age_group)
+
+    auto model1 = std::vector<mio::osecir::Model<double>>{model};
+    auto model2 = std::vector<mio::osecir::Model<double>>{model};
+    auto model3 = std::vector<mio::osecir::Model<double>>{model};
+
+    auto read_result1 = mio::osecir::read_input_data_county(
+        model1, {2020, 12, 01}, {1002}, std::vector<double>(size_t(num_age_groups), 1.0), 1.0, TEST_DATA_DIR, 10);
+
+    auto read_result2 = mio::osecir::read_input_data(
+        model2, {2020, 12, 01}, {1002}, std::vector<double>(size_t(num_age_groups), 1.0), 1.0, TEST_DATA_DIR, 10);
+
+    auto read_result_district =
+        mio::osecir::read_input_data(model3, {2020, 12, 01}, {1002}, std::vector<double>(size_t(num_age_groups), 1.0),
+                                     1.0, mio::path_join(TEST_DATA_DIR, "pydata/District"), 10);
+
+    ASSERT_THAT(read_result1, IsSuccess());
+    ASSERT_THAT(read_result2, IsSuccess());
+    ASSERT_THAT(read_result_district, IsSuccess());
+
+    // values were generated by the tested function; can only check stability of the implementation, not correctness
+    auto expected_values =
+        (Eigen::ArrayXd(num_age_groups * Eigen::Index(mio::osecir::InfectionState::Count)) << 10280, 2.82575, 1.25589,
+         0, 5.85714, 0, 1.42857, 1.33333, 28.2857, 0, 19091.3, 6.59341, 4.23862, 0, 11, 0, 2.54286, 1.33333, 88, 0,
+         73821.7, 41.9152, 21.6641, 0, 54.5714, 0, 22, 1.33333, 523.857, 0, 82533.6, 39.5604, 21.0361, 0, 51.4286, 0,
+         15.1714, 1.33333, 468.571, 0, 43733.1, 8.32025, 5.18053, 0, 11.4286, 0, 4.22857, 1.33333, 158.571, 8, 15642.7,
+         10.5181, 3.2967, 0, 3.28571, 0, 0.657143, 1.33333, 49.8571, 7.57143)
+            .finished();
+
+    ASSERT_THAT(print_wrap(model1[0].populations.array().cast<double>()),
+                MatrixNear(print_wrap(model2[0].populations.array().cast<double>()), 1e-5, 1e-5));
+
+    ASSERT_THAT(print_wrap(model1[0].populations.array().cast<double>()),
+                MatrixNear(print_wrap(model3[0].populations.array().cast<double>()), 1e-5, 1e-5));
+
+    ASSERT_THAT(print_wrap(model1[0].populations.array().cast<double>()),
+                MatrixNear(print_wrap(expected_values), 1e-5, 1e-5));
+    ASSERT_THAT(print_wrap(model2[0].populations.array().cast<double>()),
+                MatrixNear(print_wrap(expected_values), 1e-5, 1e-5));
+    ASSERT_THAT(print_wrap(model3[0].populations.array().cast<double>()),
+                MatrixNear(print_wrap(expected_values), 1e-5, 1e-5));
+}
+
+TEST(TestOdeSecir, export_time_series_init)
+{
+    TempFileRegister temp_file_register;
+    auto tmp_results_dir = temp_file_register.get_unique_path();
+    ASSERT_THAT(mio::create_directory(tmp_results_dir), IsSuccess());
+
+    constexpr auto num_age_groups = 6; //reading data requires RKI data age groups
+    mio::osecir::Model<double> model(make_model(num_age_groups));
+
+    // Test exporting time series
+    std::vector<mio::osecir::Model<double>> models{model};
```

Comment:
It would be nice to test a vector with more than one entry somewhere.

Reply:
since the casedata only has entries for county id 1002 and the reason, we are just using tested functionalities, i think one entry might be enough.

Reply:
fine